eth/v1alpha1/beacon_chain.proto

```diff
@@ -0,0 +1,338 @@
+// Copyright 2019 Prysmatic Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+syntax = "proto3";
+
+package ethereum.eth.v1alpha1;
+
+import "google/api/annotations.proto";
+import "google/protobuf/empty.proto";
+
+import "eth/v1alpha1/attestation.proto";
+import "eth/v1alpha1/block.proto";
+import "eth/v1alpha1/validator.proto";
+
+option csharp_namespace = "Ethereum.Eth.v1alpha1";
+option go_package = "github.com/prysmaticlabs/ethereumapis/eth/v1alpha1;eth";
+option java_multiple_files = true;
+option java_outer_classname = "BeaconChainProto";
+option java_package = "org.ethereum.eth.v1alpha1";
+option php_namespace = "Ethereum\\Eth\\v1alpha1";
+
+// Beacon chain API
+//
+// The beacon chain API can be used to access data relevant to the Ethereum 2.0
+// phase 0 beacon chain.
+service BeaconChain {
+    // TODO(preston): Batch requests?
+
+    // Retrieve attestations by block root, slot, or epoch. 
+    // 
+    // The server may return an empty list when no attestations match the given 
+    // filter criteria. This RPC should not return NOT_FOUND. Only one filter 
+    // criteria should be used.
+    rpc ListAttestations(ListAttestationsRequest) returns (ListAttestationsResponse) {
+        option (google.api.http) = {
+            get: "/eth/v1alpha1/beacon/attestations"
+        };
+    }
+
+    // Retrieve blocks by root, slot, or epoch. 
+    // 
+    // The server may return multiple blocks in the case that a slot or epoch is
+    // provided as the filter criteria. The server may return an empty list when
+    // no blocks in their database match the filter criteria. This RPC should 
+    // not return NOT_FOUND. Only one filter criteria should be used.
+    rpc ListBlocks(ListBlocksRequest) returns (ListBlocksResponse) {
+        option (google.api.http) = {
+            get: "/eth/v1alpha1/beacon/blocks"
+        };
+    }
+
+    // Retrieve information about the head of the beacon chain from the view of
+    // the beacon chain node. 
+    // 
+    // This includes the head block slot and root as well as information about
+    // the most recent finalized and justified slots.
+    rpc GetChainHead(google.protobuf.Empty) returns (ChainHead) {
```

Comment:
Why Get? Why not just ChainHead? Won't this generate go files that have the Get prefix, making things go against the go style guide?

Reply:
https://cloud.google.com/apis/design/standard_methods

Reply:
Yes, it might make for go method names that violate best practices, but these definitions are language agnostic so we shouldn't tailor them for go specifically.

Reply:
Great, agreed